extensions/amp-video/0.1/amp-video.js

```diff
@@ -622,13 +631,56 @@ class AmpVideo extends AMP.BaseElement {
     return ranges;
   }
 
+  /**
+    * Called when the video is first loaded.
+    * @override
+  */
+  firstLayoutCompleted() {
+    if (this.getPlaceholder() == null) {
+      this.togglePlaceholder(false);
+    }
+    this.fadeOutBlur_();
+  }
+
   /**
    * @return {!../../../src/service/url-impl.Url}
    * @private
    */
   getUrlService_() {
     return Services.urlForDoc(this.element);
   }
+
+  /**
+    * Fades out a blurry placeholder if one currently exists.
+    * @return {boolean} whether or not there was a blurry placeholder that was
+    * faded.
+    */
+  fadeOutBlur_() {
+    const placeholder = this.getPlaceholder();
+    // checks for the existance of a visible blurry placeholder
+    if (placeholder && placeholder.classList.contains('i-amphtml-blur') &&
+      isExperimentOn(this.win, 'blurry-placeholder')) {
+      setImportantStyles(placeholder, {'opacity': 0.0});
+    } else {
+      this.togglePlaceholder(false);
+    }
+  }
+
+  /**
+   * Adds an onloading event for the actual poster element for the case where
+   * the poster has a blurry image placeholder and loads before the video
+   * element.
+   * @param {*function()} opacityChangeFunc
+   * @private
+   */
```

Comment:
Where I got this logic:
https://stackoverflow.com/questions/18571424/how-to-add-event-listener-to-html5-video-poster-image-load-event
https://stackoverflow.com/questions/33799763/event-on-video-poster-loaded?noredirect=1&lq=1